pkg/cspc/algorithm/build_csp.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2020 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package algorithm
+
+import (
+	"github.com/openebs/maya/pkg/version"
+	"github.com/pkg/errors"
+	cstor "github.com/openebs/api/pkg/apis/cstor/v1"
+	"github.com/openebs/api/pkg/apis/types"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+)
+
+const (
+	// StoragePoolKindCSPC holds the value of CStorPoolCluster
+	StoragePoolKindCSPC = "CStorPoolCluster"
+	ApiVersion= "cstor.openebs.io/v1"
+)
+
+// GetCSPSpec returns a CSPI spec that should be created and claims all the
+// block device present in the CSPI spec
+func (ac *Config) GetCSPSpec() (*cstor.CStorPoolInstance, error) {
+	poolSpec, nodeName, err := ac.SelectNode()
+	if err != nil || nodeName == "" {
+		return nil, errors.Wrap(err, "failed to select a node")
+	}
+
+	// ToDo: Move following to mutating webhook.
+	if poolSpec.PoolConfig.Resources == nil {
+		poolSpec.PoolConfig.Resources = ac.CSPC.Spec.DefaultResources
+	}
+	if poolSpec.PoolConfig.AuxResources == nil {
+		poolSpec.PoolConfig.AuxResources = ac.CSPC.Spec.DefaultAuxResources
+	}
+	if len(poolSpec.PoolConfig.Tolerations) == 0 {
+		poolSpec.PoolConfig.Tolerations = ac.CSPC.Spec.Tolerations
+	}
+
+	if poolSpec.PoolConfig.PriorityClassName == "" {
+		poolSpec.PoolConfig.PriorityClassName = ac.CSPC.Spec.DefaultPriorityClassName
+	}
+
+	cspiLabels := ac.buildLabelsForCSPI(nodeName)
+
+	cspiObj := cstor.NewCStorPoolInstance().
+		WithName(ac.CSPC.Name + "-" + rand.String(4)).
+		WithNamespace(ac.Namespace).
+		WithNodeSelectorByReference(poolSpec.NodeSelector).
+		WithNodeName(nodeName).
```

Comment:
It is saying WithNodeName but it is really setting HostName.

Reply:
https://github.com/openebs/api/blob/b896c9073fd4c29042a2da9148512b260a0eace2/pkg/apis/cstor/v1/cstorpoolinstancebuilder.go#L97